util/signing/signer/src/cryptography/secp256k1.rs

```diff
@@ -0,0 +1,39 @@
+use crate::cryptography::Curve;
+use crate::{Bytes, PublicKey, Signature, VerifierError, VerifierOperations};
+use anyhow::Context;
+use k256::ecdsa::{self, VerifyingKey};
+use k256::pkcs8::DecodePublicKey;
+use ring_compat::signature::Verifier;
+
+/// The secp256k1 elliptic curve.
+#[derive(Debug, Clone)]
+pub struct Secp256k1;
+
+impl Curve for Secp256k1 {}
+
+/// Built-in verifier for secp256k1.
+#[async_trait::async_trait]
+impl VerifierOperations<Secp256k1> for Secp256k1 {
+	async fn verify(
+		&self,
+		message: Bytes,
+		signature: Signature,
+		public_key: PublicKey,
+	) -> Result<bool, VerifierError> {
+		let verifying_key = VerifyingKey::from_public_key_der(&public_key.0 .0)
```

Comment:
Any particular reason to use DER format here?
I think this just adds overhead, the byte representation as provided by rust-crypto is pretty canonical.

Reply:
`der` is what we got from `AwsKms` which is where this snippet was grabbed from.

Reply:
I see. I think, to avoid any provider specifics seeping into the general data representations, we need to agree on which representation we use natively. SEC1 is more "natural" as it does not require parsing to construct the working values, just encoding/decoding large integers from/to byte sequences, and the encoding length is exactly the bit width of the EC elements.

The AWS implementation would need to encode and decode, then.

Reply:
Yeah, so this kind of thing is why a set of `Marshalling` traits, different associated types by provider, etc. might be useful.

Reply:
Until we have a reason to have differently marshalled key and signature representations among the data types the client API needs to deal with, the specific conversions could be left to the provider implementations.